src/SampleWebApp/Controllers/AuthorssApiController.cs

```diff
@@ -0,0 +1,149 @@
+
+        public AuthorssApiController(AuthorDataContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/AuthorssApi
+        [HttpGet]
+        public IEnumerable<Author> GetAuthor()
+        {
+            return _context.Author;
+        }
+
+        // GET: api/AuthorssApi/5
+        [HttpGet("{id}", Name = "GetAuthor")]
+        public IActionResult GetAuthor([FromRoute] int id)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            Author author = _context.Author.SingleOrDefault(m => m.AuthorID == id);
+
+            if (author == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(author);
+        }
+
+        // PUT: api/AuthorssApi/5
+        [HttpPut("{id}")]
+        public IActionResult PutAuthor(int id, [FromBody] Author author)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (id != author.AuthorID)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(author).State = EntityState.Modified;
+
+            try
+            {
+                _context.SaveChanges();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!AuthorExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return new StatusCodeResult(StatusCodes.Status204NoContent);
+        }
+
+        // POST: api/AuthorssApi
+        [HttpPost]
+        public IActionResult PostAuthor([FromBody] Author author)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            _context.Author.Add(author);
+            try
+            {
+                _context.SaveChanges();
+            }
+            catch (DbUpdateException)
+            {
+                if (AuthorExists(author.AuthorID))
+                {
+                    return new StatusCodeResult(StatusCodes.Status409Conflict);
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return CreatedAtRoute("GetAuthor", new { id = author.AuthorID }, author);
+        }
+
+        // DELETE: api/AuthorssApi/5
+        [HttpDelete("{id}")]
+        public IActionResult DeleteAuthor(int id)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            Author author = _context.Author.SingleOrDefault(m => m.AuthorID == id);
+            if (author == null)
+            {
+                return NotFound();
+            }
+
+            _context.Author.Remove(author);
```

Comment:
The async / sync is based on user input. The scaffolding dialog has a checkbox which passes in a flag for this. 

In this case I happened to not pass in that flag.